### PR TITLE
Fix bad auth header error

### PR DESCRIPTION
**Pull request by grantleehoffman**

### CODE REVIEW

File: service/handlers.go [hunk 1]
```diff
@@ -150,7 +150,11 @@ func (h handler) createWorkflowFromGit(w http.ResponseWriter, r *http.Request) {
 
 	level.Debug(l).Log("message", "validating authorization header for create workflow from git")
 	ah := r.Header.Get("Authorization")
-	a := credentials.NewAuthorization(ah)
+	a, err := credentials.NewAuthorization(ah)
```
Comment (szab100) [inline service/handlers.go:153]: is it possible that the NPE is happening here? I just checked this code is calling strings.SplitN on the header value, and I'm not sure if that handles when the string is 'nil' (when no Authorization header is present on the request)?

File: service/internal/credentials/vault_test.go [hunk 2]
```diff
@@ -506,6 +506,39 @@ func TestValidateAuthorizedAdmin(t *testing.T) {
...
+			header:       "vault:testkey:testsecret",
+			expectedAuth: &Authorization{"vault", "testkey", "testsecret"},
+			expectErr:    false,
+		},
+		{
+			name:      "invalid authorization header",
+			header:    "vault:testbad",
+			expectErr: true,
+		},
+	}
```
Comment (szab100) [inline service/internal/credentials/vault_test.go:527]: can we add a test case where 'header' is 'nil'?

**Review by szab100 — COMMENTED** [review]
Thank you

**Review by szab100 — APPROVED** [review]
LGTM, thanks!
(sorry I did not realize it is 'string', therefore cannot be 'nil'.)

File: service/handlers.go (hunk 1 above)
Comment (grantleehoffman) [reply service/handlers.go:153]: Talked through this, should be handled
